sklearn/model_selection/tests/test_validation.py

```diff
@@ -936,6 +937,124 @@ def test_cross_val_predict_with_method():
         assert_array_almost_equal(expected_predictions, predictions)
 
 
+def test_cross_val_predict_corner_case():
```

Comment:
Although we are modifying a subset of our samples, still somehow I am unable to relate 'class subset' to this test representing a rare case. Over 'corner case' though, 'edge case' seemed better in brevity.